### ISSUE TITLE
Detection model training questions

### ISSUE THREAD

**Issue by zhiltsov-max** [open-edge-platform/training_extensions#912]
Hi, I'm trying to train a detection model on a custom COCO-like dataset (basically, it's a subsampled VOC with only `car` and  `person` classes converted to COCO, as other formats are not supported by OTE). I'm trying to use a model and follow the tutorial from the quick start guide.

Few questions:
1. It's not clear if I have to change anything in the model, if my dataset has different number of classes than COCO
1. The [ATSS model](https://github.com/openvinotoolkit/mmdetection/tree/ote/configs/ote/custom-object-detection/gen3_mobilenetV2_ATSS) training is always diverges during the 1st epoch. Tried learning rates from 0.1 to 0.00001. Also, in the logs I see it has loss bursts each ~6 iterations: if "normally" it has from 1.5 to 4, bursts have from 50 to 200.
2. The SSD model seem to converge successfully with the same parameters. However, as I understand from the table [here](https://github.com/openvinotoolkit/mmdetection/tree/825488ada5e07f8461c991ab37578bab5eedf423/configs/ote/custom-object-detection) it should support multiple GPU training, but only 1 is used (I have 2 equal).
4. Probably, it would be useful to support learning rate decrease instead of training with a fixed one.
5. Probably, it would be useful to include the template used in the output model directory. This information is easy to get lost and it is needed in all the commands.
6. After exporting, I've got the following value in `confidence_threshold`: `333?` (supposed to be ~0.7)

**Comment by morkovka1337** [comment]
Hi, @zhiltsov-max,

1. You have to change the field `num_classes` in the `bbox_head` in the `model.py` file
2. Try to repeat the training with the changes from the step above. If the behavior reproduces, please, share the content of the config file (`model.py` by default).
3. If you run the training using `python tools/train.py` (in the mmdetection folder), it runs single gpu training. In order to run distributed training, use `tools/dist_train.sh <params>`.
4. You can change the learning rate policy via changing `lr_config` value. See [here](https://github.com/openvinotoolkit/mmdetection/blob/ote/docs/tutorials/customize_runtime.md#customize-training-schedules) for details.
5. Sorry, I did not get the suggestion :( You mean add model template file to the outputs folder? Usually, for every experiment a separate model is created with a separate config. Thus, you have an outputs folder corresponding to a specific configuration.
6. That looks strange, could you, please, share the details?

**Comment by zhiltsov-max** [comment]
Thank you for the responses. Basically, I just follow the https://github.com/openvinotoolkit/training_extensions/blob/master/QUICK_START_GUIDE.md

1, 2. Ok, I'll check it. ​I just use the template from the folder.
3. 4. I don't interact with separate scripts, just calling `ote train`: 
```bash
ote train training_extensions/external/mmdetection/configs/ote/custom-object-detection/gen3_mobilenetV2_SSD/template.yaml \
  --train-ann-files sample_dataset/annotations/instances_train.json  \
  --train-data-roots sample_dataset/images/train/ \
  --val-ann-files sample_dataset/annotations/instances_validation.json \
  --val-data-roots sample_dataset/images/validation/  \
  --save-model-to sample_dataset_model/ \
  params \
  --learning_parameters.batch_size 4 \
  --learning_parameters.learning_rate 0.005 \
  --learning_parameters.num_iters 5
```  
5. I mean, after calling the command above, in the output directory I only have 2 files
![image](https://user-images.githubusercontent.com/13832349/153588593-52bdc5b1-904b-4149-a4fd-0ccf196aae11.png)
However, all the `ote` commands require the model template argument. It is easy to loose this info during experimenting (eg. after the subrepository was updated) and it will be unrecoverable.
6. This is what I've got after calling
```bash
ote export \
  --load-weights sample_dataset_model/weights.pth \
  --save-model-to sample_dataset_model-ov \
  training_extensions/external/mmdetection/configs/ote/custom-object-detection/gen3_mobilenetV2_SSD/template.yaml
```
In a hex editor I can see it can be parsed as 0.69(9), so I suppose, it can be a binary representation of a number. I suggest to change it to a string, because the binary form it's not portable between platforms and less convenient for users.

**Comment by druzhkov-paul** [comment]
@zhiltsov-max 

1. You don't have to change anything in the model itself, if you use it via OTE CLI. Number of classes is automatically adjusted according to the information about the labels in the annotation file. Please see label schema for the classes that were actually extracted from it.
2. If you find the model with a default training setup to diverge on your dataset, please give us some more information about your specific case. I believe you can contact us directly for this.
3. All models are currently set up to be trained on a single GPU. The numbers from the table you are referring to are for pre-training, that is already done and should not be cared much by the user.
4. All training schedules were carefully selected though an extensive ablation study.
5. Thank you for suggestion.
6. Yes, this is a binary file. Users should not care about the content of internal representation files.

**Comment by sungmanc** [comment]
Close this issue, please reopen it if there is any problem.
